hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/impl/prefetch/SingleFilePerBlockCache.java

```diff
@@ -310,7 +333,12 @@ public void close() throws IOException {
     int numFilesDeleted = 0;
 
     for (Entry entry : blocks.values()) {
-      entry.takeLock(Entry.LockType.WRITE);
+      boolean lockAcquired = entry.takeLock(Entry.LockType.WRITE, 5, TimeUnit.SECONDS);
```

Comment:
pull the number into a constant. Know that I automatically -1 all inline constants in production code and save time all round.

Reply:
my bad, let me fix this real quick